rotkehlchen/api/rest.py

```diff
@@ -4037,3 +4042,166 @@ def query_kraken_staking_events(
+                except UnknownAsset as e:
+                    log.error(e)
+        except PermissionError as e:
+            return wrap_in_fail_result(
+                message=f'Failed to create information file. {str(e)}',
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            )
+        except sqlcipher.Error as e:  # pylint: disable=no-member
+            return wrap_in_fail_result(
+                message=f'Failed to retrieve the list of assets {str(e)}',
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            )
+        cursor = GlobalDBHandler()._conn.cursor()
+        query = cursor.execute('SELECT value from clean_db.settings WHERE name=="version";')
+        version = query.fetchone()[0]
+        data = {
+            'version': version,
+            'assets': serialized,
+        }
+
+        zip_path = dirpath / 'assets.zip'
+        with ZipFile(zip_path, 'w') as assets_zip:
+            assets_zip.writestr(
+                zinfo_or_arcname='assets.json',
+                data=json.dumps(data),
+            )
+
+        return _wrap_in_ok_result({'file': str(zip_path)})
+
+    @require_loggedin_user()
+    def get_user_added_assets(self, async_query: bool) -> Response:
+        if async_query:
+            return self._query_async(
+                command='_get_user_added_assets',
+            )
+
+        response = self._get_user_added_assets()
+        result = response['result']
+        msg = response['message']
+        status_code = _get_status_code_from_async_response(response)
+        if result is None:
+            return api_response(wrap_in_fail_result(msg), status_code=status_code)
+
+        # success
+        result_dict = _wrap_in_result(result, msg)
+        return api_response(result_dict, status_code=status_code)
+
+    def _import_user_assets(self, path: Path) -> Dict[str, Any]:
+        globaldb = GlobalDBHandler()
+
+        try:
+            with open(path) as f:
+                data = ExportedAssetsSchema().loads(f.read())
+        except ValidationError as e:
+            return wrap_in_fail_result(
+                message=f'Provided file does not have the expected format. {str(e)}',
+                status_code=HTTPStatus.BAD_REQUEST,
+            )
+
+        # First assert that the version is correct
+        if int(data['version']) != GLOBAL_DB_VERSION:
+            return wrap_in_fail_result(
+                message='Provided file is for a different version of rotki',
+                status_code=HTTPStatus.BAD_REQUEST,
+            )
+
+        identifiers = []
+        for asset_data in data['assets']:
+            # Check if we already have the asset with that name and symbol. It is possible that
+            # we have added a missing asset
+            asset_type = AssetType.deserialize_from_db(asset_data['asset_type'])
+            identifier = globaldb.check_asset_exists(
+                asset_type=asset_type,
```

Comment:
Hmmm but there can be multiple assets of the same type (say ERC20) with same name and symbol. When possible shouldn't you check the unique identifiers, say the ethereum_address?

Reply:
This functions check for this, by checking the identifier

Reply:
Which functions?

Reply:
`check_asset_exists` in the globaldb https://github.com/rotki/rotki/blob/bf22cc735ddccd1a3f125797b5d382911d54ba08/rotkehlchen/globaldb/handler.py#L466

Reply:
No it does not. Read the function and tell me what it will return if two ERC20 tokens with the same name and symbol are in the DB and you try to use this function to see if one exists.

Reply:
Filtering assets with this logics seems good to me. What I'll do is if it's ethereum token check the identifier otherwise use this